NATS.Client.Core/Internal/NuidWriter.cs

```diff
@@ -75,23 +85,47 @@ private static bool TryWriteNuidCore(Span<char> buffer, Span<char> prefix, ulong
 
     private static uint GetIncrement()
     {
+#if NET6_0_OR_GREATER
         return (uint)Random.Shared.Next(MinIncrement, MaxIncrement + 1);
+#else
+        return (uint)new Random((int)DateTime.UtcNow.Ticks).Next(MinIncrement, MaxIncrement + 1);
+#endif
     }
 
     private static ulong GetSequential()
     {
+#if NET6_0_OR_GREATER
         return (ulong)Random.Shared.NextInt64(0, (long)MaxSequential + 1);
+#else
+        var max = MaxSequential + 1;
+        var rand = new Random((int)DateTime.UtcNow.Ticks);
```

Comment:
```suggestion
        var rand = new Random();
```

The parameterless overload is better than seeding with the current time.